packages/core/src/components/Form.tsx

```diff
@@ -804,14 +807,15 @@ export default class Form<
     }
   }
 
-  /** Programmatically validate the form. If `onError` is provided, then it will be called with the list of errors the
-   * same way as would happen on form submission.
+  /** Validates the form using the given `formData`. For use on form submission or on programmatic validation.
+   * If `onError` is provided, then it will be called with the list of errors.
    *
+   * @param formData - The form data to validate
    * @returns - True if the form is valid, false otherwise.
    */
-  validateForm() {
+  validateFormWithFormData = (formData: T | undefined): boolean => {
```

Comment:
Making the prop optional via the `?` notation has the same effect and is preferred
```suggestion
  validateFormWithFormData = (formData?: T): boolean => {
```